### PR TITLE
New: Add option for disabled button text (fixes #207)

### LINKED ISSUE #207: Alternate button text when locked
### Subject of the enhancement
When a Trickle button is locked/disabled, it will show the text from `_trickle._button.text` (if it exists). A common customization of ours is to use different button text depending on if it is locked or unlocked. For example:

- Trickle button (locked) - "Please complete the content above"
- Trickle button (unlocked) - "Click here to continue"

This functionality could be achieved by adding a new property of `textLocked` to `_trickle._button`. This would be backwards-compatible since we would not change or remote the `text` property. We would also want to add `ariaLabelLocked` and similar properties relating to `startText` and `finalText`.

### PR DESCRIPTION
**Pull request by swashbuck**
Fix #207 

### New
* Adds the ability to show different text when the button is **disabled**. The new properties are `disabledText` and `disabledAriaLabel`.

### Testing
1. Set up Trickle using the code from _example.json_
2. `_styleBeforeCompletion` should be set to `visible` so that you can see the button before it is enabled
3. Test different scenarios including:

- `text` is set but `disabledText` is not set.
- `text` and `disabledText` are set as well as `startText` and `finalText`

The order of precedence that text will be shown is:

`disabledText`
`startText`
`finalText`
`text`

For instance, if the Trickle is on the final block and `disabledText` is set, that will be used. Otherwise, it will check for `finalText` when the button is disabled. Finally, it will fall back to `text`.

### CODE REVIEW

**Comment by joe-allen-89** [comment]
It's working as described, it feels a bit strange that throughout the page trickle says it's locked but on the last block it doesn't if finalText is set despite it being locked, might be confusing to users.

**Comment by swashbuck** [comment]
> It's working as described, it feels a bit strange that throughout the page trickle says it's locked but on the last block it doesn't if finalText is set despite it being locked, might be confusing to users.

@joe-allen-89 Maybe we should add locked text for final and start as well? We never use those options in projects, as far as I know.

**Comment by oliverfoster** [comment]
I'd prefer you called it `disabledText` and `disabledAriaLabel`. As locking is a model state and disabled is a button state. The convention is to put the descriptor first - `playText`, `stopText`, `itemText`, `resumeButtonText`, `promptText`, `buttonText`, `startText`, `finalText`, `backLabel`, `nextLabel` etc.

**Comment by oliverfoster** [comment]
> > It's working as described, it feels a bit strange that throughout the page trickle says it's locked but on the last block it doesn't if finalText is set despite it being locked, might be confusing to users.
> 
> @joe-allen-89 Maybe we should add locked text for final and start as well? We never use those options in projects, as far as I know.

Maybe it should just take precedence over `startText` and `finalText`?

**Comment by swashbuck** [comment]
@joe-allen-89 @oliverfoster Changes made and PR description updated

**Review by oliverfoster — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -70,8 +70,12 @@ The attributes listed below are properly formatted as JSON in [*example.json*](h
 
 >>**text** (string):  Defines the default button text. The default is `"Continue"`.
 
+>>**disabledText** (string):  Defines the default button text when the button is disabled. If not set, `text`, `startText`, or `finalText` will be used. `_styleBeforeCompletion` must be set to `visible`. The default is `""`.
```
Comment (oliverfoster) [inline README.md:73]: Please rewrite the latter options in order of precedence, the final paragraph of the pr body needs changing in the same regard also.

**Review by oliverfoster — APPROVED** [review]

**Review by joe-allen-89 — APPROVED** [review]
👍

File: schema/article.schema.json [hunk 2]
```diff
@@ -126,6 +126,15 @@
                     "translatable": true
                   }
                 },
+                "disabledText": {
```
Comment (cahirodoherty-learningpool) [inline schema/article.schema.json:129]: I think this should come _after_ `ariaLabel` as that is related to the button above

Comment (swashbuck) [reply schema/article.schema.json]: Agreed. See [b21fbda](https://github.com/adaptlearning/adapt-contrib-trickle/pull/214/commits/b21fbdaba66024a17d2f6f1845f8c64f8c72e62d).

File: README.md [hunk 3]
```diff
@@ -70,8 +70,12 @@ The attributes listed below are properly formatted as JSON in [*example.json*](h
 
 >>**text** (string):  Defines the default button text. The default is `"Continue"`.
 
+>>**disabledText** (string):  Defines the default button text when the button is disabled. If not set, `finalText`, `startText` or `text` will be used in that order. `_styleBeforeCompletion` must be set to `visible` so that the button is shown while disabled. The default is `""`.
```
[the remark this answers is not in the document]
Comment (swashbuck) [reply README.md]: Fixed in [f627fa1](https://github.com/adaptlearning/adapt-contrib-trickle/pull/214/commits/f627fa19072b02f800ce7c4958957770ed94955b).

**Review by cahirodoherty-learningpool — APPROVED** [review]
👀

**Pull request merged**